pkg/container/network.go

```diff
@@ -53,35 +74,82 @@ func SetupContainerNetworking() ([]DHCPInterface, error) {
 	})
 
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return dhcpIfaces, nil
+	return fcIfaces, dhcpIfaces, nil
 }
 
-func networkSetup(dhcpIfaces *[]DHCPInterface) (bool, error) {
-	ifaces, err := net.Interfaces()
-	if err != nil || ifaces == nil || len(ifaces) == 0 {
+func filterIgnored(allIfaces []net.Interface, extraIntfs map[string]struct{}) (result []net.Interface) {
+
+	for _, intf := range allIfaces {
+
+		// first process explicitly ignored
+		if _, ok := ignoreInterfaces[intf.Name]; ok {
+			continue
+		}
+
+		// next process extra intfs
+		if _, ok := extraIntfs[intf.Name]; ok {
+			result = append(result, intf)
+		}
+
+		// add intfs with IPs
+		addrs, _ := intf.Addrs()
+		if len(addrs) > 0 {
+			result = append(result, intf)
+		}
+
+	}
+
+	return result
+}
+
+func networkSetup(fcIfaces *firecracker.NetworkInterfaces, dhcpIfaces *[]DHCPInterface, extraIntfs map[string]struct{}, expectedIntfNum *int) (bool, error) {
+	allIfaces, err := net.Interfaces()
+	if err != nil || allIfaces == nil || len(allIfaces) == 0 {
 		return true, fmt.Errorf("cannot get local network interfaces: %v", err)
 	}
 
-	// interfacesCount counts the interfaces that are relevant to Ignite (in other words, not ignored)
-	interfacesCount := 0
+	ifaces := filterIgnored(allIfaces, extraIntfs)
+	if len(ifaces) < *expectedIntfNum {
+		return true, fmt.Errorf("not enough extra interfaces connected (%d/%d), waiting", len(ifaces), *expectedIntfNum)
+	}
+
+	// Sorting interfaces to make sure eth0 is always first
+	sort.Slice(ifaces, func(i, j int) bool {
+		return ifaces[i].Name == mainInterface
+	})
+
 	for _, iface := range ifaces {
-		// Skip the interface if it's ignored
-		if _, ok := ignoreInterfaces[iface.Name]; ok {
+
+		// Try to transfer the address from the container to the DHCP server
+		ipNet, gw, noIPs, err := takeAddress(&iface)
+
+		// If interface has no IPs configured, setup tc redirect
+		if noIPs && iface.Name != mainInterface {
+			log.Printf("Interface %s has no IP, setting up tc redirect", iface.Name)
+			tcInterface, err := addTcRedirect(&iface)
```

Comment:
Maybe an explicit opt-in to the tc-redirect interface mode can eliminate the need for a heuristic check here.
It may make it easier to differentiate from `macvtap` and support future, esoteric, interface modes.

Does that sound like a good decision to you?

Example API from https://github.com/weaveworks/ignite/pull/836#discussion_r633692902:
```yaml
metadata:
  annotations:
    ignite.weave.works/interface-mode/eth0: 'dhcp-bridge'
    ignite.weave.works/interface-mode/eth2: 'tc-redirect'
    ignite.weave.works/interface-mode/eth3: 'tc-redirect'
    ignite.weave.works/interface-mode/macvtap-uvnrt: 'macvtap'
    ignite.weave.works/interface-mode/macvtap-hgfeq: 'macvtap'
```

The key format can be whatever makes sense to you.
`ignite.weave.works/interface/eth1` or `ignite.weave.works/interface/eth1/mode` could be good alternatives.
This allows us to add expected behavior about each interface without parsing a custom CSV+K=V format.

Parsing would be a similar iteration with prefix/suffix filtering, similar to the sandbox-env suggestion.

Reply:
ok, yeah, makes sense.